python/ray/experimental/workflow/storage/s3.py

```diff
@@ -224,6 +95,28 @@ async def _list_objects(self, path: Optional[str] = None) -> List[str]:
                     workflow_ids.append(prefix)
         return workflow_ids
 
+    @data_load_error
+    async def list_workflow(self) -> List[str]:
+        return await self._list_objects()
+
+    @data_load_error
+    async def get_step_status(self, workflow_id: str,
```

Comment:
These two APIs make the DB complicated. How about this:
```
async def scan_prefix(self, path) -> str
```

Then `workflow_storage` can just use this API to get_status.